.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,47 @@
+# Github actions that run tests on each commit, and publish to PyPi on tagged releases
+# build using https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+
+name: Test and Release to PyPi
+
+on:
+  push
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install and build
+      run: >-
+        python -m
+        pip install ".[dev]"
+        build
+        --user
+    - name: build binary and tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+    - name: test
+      run: >-
+        python -m
+        pip install --upgrade pytest
+        pytest
+    - name: Publish distribution to test pypi
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+    - name: Publish distribution to PyPI
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.PYPI_TOKEN }}
```

Comment:
From https://github.com/pypa/gh-action-pypi-publish, it looks like `user` needs to be `__token__` .Does this work without that?

Reply:
Conflicting messaging int he pypi docs.
The one I was referring to - https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ - seems to say it's fine without, but what you linked looks more recent, will change to that.